common/utils.h

```diff
@@ -0,0 +1,68 @@
+/*
+  utils.h
+
+  This file is part of GammaRay, the Qt application inspection and
+  manipulation tool.
+
+  Copyright (C) 2010-2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Filipe Azevedo <filipe.azevedo@kdab.com>
+
+  Licensees holding valid commercial KDAB GammaRay licenses may use this file in
+  accordance with GammaRay Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+  @file
+  This file is part of the GammaRay Common API and declares various utility
+  methods needed when writing shared code.
+
+  @brief
+  Declares various utility methods needed when writing shared code.
+
+  @author Filipe Azevedo \<filipe.azevedo@kdab.com\>
+*/
+
+#ifndef GAMMARAY_COMMON_UTILS_H
+#define GAMMARAY_COMMON_UTILS_H
+
+#include "gammaray_common_export.h"
+
+#include <QColor>
+
+namespace GammaRay {
+/**
+ * @brief GammaRay Common utilities.
+ */
+namespace Utils {
+/**
+ * Returns true if the given color is ligth, else false.
+ * @param color is the color to check.
+ *
+ * @return a bool telling if the given color is light or not.
+ */
+GAMMARAY_COMMON_EXPORT bool isLightColor(const QColor &color);
+/**
+ * Returns true if the given color is dark, else false.
+ * @param color is the color to check.
+ *
+ * @return a bool telling if the given color is dark or not.
+ */
+GAMMARAY_COMMON_EXPORT bool isDarkColor(const QColor &color);
```

Comment:
Quite possible indeed, but I think we should rather link the tests to gammaray_ui in that case.

We have gotten requests from people wanting to use GammaRay on a QtCore-only device. That is currently not possible, but I'm at least trying to not move into the opposite direction with changes we make to the probe :)

Independent of that, it's a bit safer to not offer methods to the probe that might impact visual output on the client, somebody someday might use them ;-) We had quite some trouble in the past with dark color schemes on the client and light color schemes on the target device making the output unreadable.